### PR TITLE
Job definition resources unit tests

### PR DESCRIPTION
**Pull request by surajkota**
Description of changes:
Resources covered in this PR:
- ModelQualityJobDefinition
  - Coverage:
    - total pkg: 60.6%
    - sdk.go: 73%
    - manager.go: 53.4%
- ModelExplainabilityJobDefinition
  - Coverage:
    - total pkg: 61.6%
    - sdk.go: 76.1%
    - manager.go: 51.9%
- DataQualityJobDefinition
  - Coverage:
    - total pkg: 59%
    - sdk.go: 72.2%
    - manager.go: 51.9%
- ModelBiasJobDefinition
  - Coverage:
    - total pkg: 63.6%
    - sdk.go: 78.2%
    - manager.go: 51.9%

Overall sagemaker controller pkg coverage: 38%

Testing:
```
make test
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/data_quality_job_definition/testdata/test_suite.yaml [hunk 1]
```diff
@@ -2,16 +2,119 @@ tests:
...
+      given:
+        desired_state: "v1alpha1/create/observed/success_after_create.yaml"
+        svc_api:
+          - operation: DescribeDataQualityJobDefinitionWithContext
+            output_fixture: "sdkapi/describe/success_describe.json"
+      invoke: ReadOne
+      expect:
+        latest_state: "v1alpha1/create/observed/success_after_create.yaml"
+    - name: "ReadOne=SuccessClearsConditions"
+      description: Testing a successful reconiliation clears conditions if terminal/recoverable condition were already set to true
```
Comment (ryansteakley) [inline pkg/resource/data_quality_job_definition/testdata/test_suite.yaml:75]: nit: typo should be `reconciliation`

File: pkg/resource/model_bias_job_definition/testdata/test_suite.yaml [hunk 2]
```diff
@@ -2,16 +2,119 @@ tests:
...
+        given:
+          desired_state: "v1alpha1/create/observed/success_after_create.yaml"
+          svc_api:
+            - operation: DescribeModelBiasJobDefinitionWithContext
+              output_fixture: "sdkapi/describe/success_describe.json"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/create/observed/success_after_create.yaml"
+      - name: "ReadOne=SuccessClearsConditions"
+        description: Testing a successful reconiliation clears conditions if terminal/recoverable condition were already set to true
```
Comment (ryansteakley) [inline pkg/resource/model_bias_job_definition/testdata/test_suite.yaml:75]: same nit

File: pkg/resource/model_explainability_job_definition/testdata/test_suite.yaml [hunk 3]
```diff
@@ -2,16 +2,119 @@ tests:
...
+      given:
+        desired_state: "v1alpha1/create/observed/success_after_create.yaml"
+        svc_api:
+          - operation: DescribeModelExplainabilityJobDefinitionWithContext
+            output_fixture: "sdkapi/describe/success_describe.json"
+      invoke: ReadOne
+      expect:
+        latest_state: "v1alpha1/create/observed/success_after_create.yaml"
+    - name: "ReadOne=SuccessClearsConditions"
+      description: Testing a successful reconiliation clears conditions if terminal/recoverable condition were already set to true
```
Comment (ryansteakley) [inline pkg/resource/model_explainability_job_definition/testdata/test_suite.yaml:75]: same nit

File: pkg/resource/model_quality_job_definition/manager.go [hunk 4]
```diff
@@ -98,7 +99,11 @@ func (rm *resourceManager) ReadOne(
 		}
 		return rm.onError(r, err)
 	}
-	return rm.onSuccess(observed)
+	// return rm.onSuccess(observed)
```
Comment (ryansteakley) [inline pkg/resource/model_quality_job_definition/manager.go:102]: Should remove this code, leftover from unit-test development?

File: pkg/resource/model_quality_job_definition/manager.go [hunk 5]
```diff
@@ -117,7 +122,11 @@ func (rm *resourceManager) Create(
 	if err != nil {
 		return rm.onError(r, err)
 	}
-	return rm.onSuccess(created)
+	// return rm.onSuccess(created)
```
Comment (ryansteakley) [inline pkg/resource/model_quality_job_definition/manager.go:125]: same here

File: pkg/resource/model_quality_job_definition/manager.go [hunk 6]
```diff
@@ -167,6 +176,12 @@ func (rm *resourceManager) Delete(
 		return rm.onError(r, err)
 	}
 
+	if observed != nil {
```
Comment (ryansteakley) [inline pkg/resource/model_quality_job_definition/manager.go:179]: same

File: pkg/resource/model_quality_job_definition/testdata/test_suite.yaml [hunk 7]
```diff
@@ -2,16 +2,119 @@ tests:
...
+      given:
+        desired_state: "v1alpha1/create/observed/success_after_create.yaml"
+        svc_api:
+          - operation: DescribeModelQualityJobDefinitionWithContext
+            output_fixture: "sdkapi/describe/success_describe.json"
+      invoke: ReadOne
+      expect:
+        latest_state: "v1alpha1/create/observed/success_after_create.yaml"
+    - name: "ReadOne=SuccessClearsConditions"
+      description: Testing a successful reconiliation clears conditions if terminal/recoverable condition were already set to true
```
Comment (ryansteakley) [inline pkg/resource/model_quality_job_definition/testdata/test_suite.yaml:75]: same nit as others.

**Review by ryansteakley — APPROVED** [review]

**Review by akartsky — APPROVED** [review]

**Pull request merged**